PROTO_tests/tests/client_dtypes_test.py

```diff
@@ -0,0 +1,178 @@
+    def test_bit_vector_edge_case(self, int_types):
+        bv_answer = [
+            "||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||..",
+            "||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|",
+            "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.",
+        ]
+
+        arr = ak.arange(2**64 - 4, 2**64 - 1, dtype=int_types)
+        bv = ak.BitVector(arr, width=64)
+        assert isinstance(bv, client_dtypes.BitVector)
+        assert bv.to_list() == bv_answer
+        assert bv.dtype == ak.bitType
+
+    def test_field_creation(self):
+        values = ak.arange(4)
+        names = ["8", "4", "2", "1"]
+        f = ak.Fields(values, names)
+        assert isinstance(f, ak.Fields)
+        assert f.to_list() == ["---- (0)", "---1 (1)", "--2- (2)", "--21 (3)"]
+        assert f.dtype == ak.bitType
+
+        # Named fields with reversed bit order
+        values = ak.array([0, 1, 5, 8, 12])
+        names = ["Bit1", "Bit2", "Bit3", "Bit4"]
+        f = ak.Fields(values, names, MSB_left=False, separator="//")
+        expected = [
+            "----//----//----//----// (0)",
+            "Bit1//----//----//----// (1)",
+            "Bit1//----//Bit3//----// (5)",
+            "----//----//----//Bit4// (8)",
+            "----//----//Bit3//Bit4// (12)",
+        ]
+        assert f.to_list() == expected
+
+        values = ak.arange(8, dtype=ak.uint64)
+        names = [f"Bit{x}" for x in range(65)]
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names)
+
+        names = ["t", "t"]
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names)
+
+        names = ["t", ""]
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names)
+
+        names = ["abc", "123"]
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names, separator="abc")
+
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names)
+
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names, pad="|!~", separator="//")
+
+    @pytest.mark.parametrize("int_types", INT_TYPES)
+    def test_ipv4_creation(self, int_types):
+        ip_list = ak.array(IPS, dtype=int_types)
+        ipv4 = ak.IPv4(ip_list)
+        py_ips = [ipaddress.IPv4Address(ip).compressed for ip in IPS]
+
+        assert isinstance(ipv4, ak.IPv4)
+        assert ipv4.to_list() == py_ips
+        assert ipv4.dtype == ak.bitType
+
+        with pytest.raises(TypeError):
+            ipv4 = ak.IPv4(f"{IPS}")
+        with pytest.raises(TypeError):
+            ipv4 = ak.IPv4(ak.array([IPS + 0.177]))
+
+        # Test handling of python dotted-quad input
+        ipv4 = ak.ip_address(py_ips)
+        assert isinstance(ipv4, ak.IPv4)
+        assert ipv4.to_list() == py_ips
+        assert ipv4.dtype == ak.bitType
+
+    def test_ipv4_normalization(self):
+        ip_list = ak.array(IPS)
```

Comment:
Again, I would really like this to be run on a larger range than just 4 numbers. Could we update this similar to my previous comment to use randint or something like that?

Reply:
Updated